### PR TITLE
Move case accessors into CommCareCase.objects - part 1

### PR DESCRIPTION
**Pull request by millerdev**
Continuation of work from https://github.com/dimagi/commcare-hq/pull/31045: form accessors now done, moving on to cases.

Part 1:
- Move case models (`CommCareCase`, `CommCareCaseIndex`, `CaseAttachment`, etc.) into `corehq.form_processor.models.cases` module.
- Move `CaseAccessorSQL.get_case` to `CommCareCase.objects.get_case`

In the upcoming PRs all the remaining case accessor methods will be moved into `CommCareCase.objects` (which is `CommCareCaseManager`), and `CaseAccessor(s|SQL)` will be removed.

`CommCareCase.objects.get_case` was done in a few steps since there are many places that call it. Most other case accessor methods can be done in a single commit since they are called in fewer places.

:blowfish: Review by commit.

### Safety story

In addition to preserving all of the old tests and adapting the old methods that are being phased out to call the new method, new tests (copies of the old tests adapted to the new structure) are being added for the new model manager.

### Automated test coverage

Yes.

### QA Plan

No QA.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by dannyroberts** [comment]
Not sure what's going on here since
```
  psycopg2.OperationalError: FATAL:  database "test_commcarehq" does not exist
```
doesn't seem that related to the change?

**Comment by millerdev** [comment]
There were patches in `SimpleTestCase` tests that were no longer patching db access. Should be fixed in https://github.com/dimagi/commcare-hq/pull/31051/commits/ec0b9cee31e254ae7702597948fb457048a93916

File: corehq/form_processor/backends/sql/dbaccessors.py [hunk 1]
```diff
@@ -411,10 +411,8 @@ class CaseAccessorSQL:
 
     @staticmethod
     def get_case(case_id):
-        try:
-            return CommCareCase.objects.partitioned_get(case_id)
-        except CommCareCase.DoesNotExist:
-            raise CaseNotFound(case_id)
+        """DEPRECATED"""
```
Comment (kaapstorm) [inline corehq/form_processor/backends/sql/dbaccessors.py:414]: Would it be worth using `warn()`?
```suggestion
        warnings.warn(
            'CaseAccessorSQL.get_case() is deprecated. Please use '
            'CommCareCase.objects.get_case()',
            DeprecationWarning
        )
```
As an added bonus, PyCharm (and probably VS Code too) will display the method with a strike through, and the warning as a tooltip.

File: corehq/form_processor/interfaces/dbaccessors.py [hunk 2]
```diff
@@ -29,12 +28,8 @@ def db_accessor(self):
         return CaseAccessorSQL
 
     def get_case(self, case_id):
-        if not case_id:
-            raise CaseNotFound
-        case = self.db_accessor.get_case(case_id)
-        if case.domain != self.domain:
-            raise CaseNotFound(case_id)
-        return case
+        """DEPRECATED use CommCareCase.objects"""
```
Comment (kaapstorm) [inline corehq/form_processor/interfaces/dbaccessors.py:31]: Ditto `warn()` suggestion.

**Review by kaapstorm — APPROVED** [review]

File: corehq/form_processor/backends/sql/dbaccessors.py (hunk 1 above)
Comment (millerdev) [reply corehq/form_processor/backends/sql/dbaccessors.py:414]: I don't think so. The plan is to delete these on fairly a short time-scale, as soon as they are not used by any more code. Hopefully that will be within a week or so.

Comment (millerdev) [reply corehq/form_processor/backends/sql/dbaccessors.py:411]: https://github.com/dimagi/commcare-hq/pull/31060/commits/9b6b7d300a5b25d5c5e03736c8c1cb132ac8ec40

**Pull request merged**